src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandReportPublisher.java

```diff
@@ -281,7 +281,7 @@ private void processBuildOutput(Run build, SauceOnDemandBuildAction buildAction,
     }
 
     protected SauceREST getSauceREST(Run build) {
-        SauceCredentials credentials = build.getAction(SauceOnDemandBuildAction.class).getCredentials();
+        SauceCredentials credentials = getBuildAction(build).getCredentials();
```

Comment:
It calls below function, and it works, I am still wrapping my head around what exactly is happening here. What do you mean by sharing in this context?

``` java
private SauceOnDemandBuildAction getBuildAction(Run build) {
        SauceOnDemandBuildAction buildAction = build.getAction(SauceOnDemandBuildAction.class);
        if (buildAction == null && build instanceof MavenBuild) {
            //try the parent
            buildAction = ((MavenBuild) build).getParentBuild().getAction(SauceOnDemandBuildAction.class);
        }
        return buildAction;
    }
```